### PR TITLE
Add Pausable Modifiers to Wrapper

### PR DESCRIPTION
**Pull request by ejwessel**
## Description
https://github.com/airswap/airswap-protocols/issues/239

Quick description:

- [ ] N/A
- [X] New features
- [ ] Bug fixes
- [ ] Typo/small tweaks

## Changes
- added modifiers `paused()` and `notPaused()`
- added `setPausedStatus()`
- added `killContract()`
- added modifiers to swap() and `killContract()` 

## Tests and Coverage
```
  38 passing (11s)

--------------|----------|----------|----------|----------|----------------|
File          |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------|----------|----------|----------|----------|----------------|
 contracts/   |      100 |      100 |      100 |      100 |                |
  Imports.sol |      100 |      100 |      100 |      100 |                |
  Wrapper.sol |      100 |      100 |      100 |      100 |                |
--------------|----------|----------|----------|----------|----------------|
All files     |      100 |      100 |      100 |      100 |                |
--------------|----------|----------|----------|----------|----------------|
```

### CODE REVIEW

File: source/wrapper/contracts/Wrapper.sol [hunk 1]
```diff
@@ -55,6 +76,25 @@ contract Wrapper {
...
+  */
+  function setPausedStatus(bool newStatus) external onlyOwner {
+    contractPaused = newStatus;
+  }
+
+  /**
+    * @notice Destroy the Contract
+    * @dev Only callable by owner and when contractPaused
+    *
+    * @param recipient address Recipient of any money in the contract
```
Comment (alicevhenshaw) [inline source/wrapper/contracts/Wrapper.sol]: `money` -> `ETH`

File: source/wrapper/test/Wrapper-unit.js [hunk 2]
```diff
@@ -329,4 +331,68 @@ contract('Wrapper Unit Tests', async accounts => {
       )
     })
   })
+
+  describe('Test pausability', async () => {
+    it('Test setPausedStatus is only callable by owner', async () => {
+      await reverted(wrapper.setPausedStatus(true, { from: notOwner }))
```
Comment (alicevhenshaw) [inline source/wrapper/test/Wrapper-unit.js]: maybe check the revert message is `        'Ownable: caller is not the owner'`

File: source/wrapper/contracts/Wrapper.sol [hunk 3]
```diff
@@ -55,6 +76,25 @@ contract Wrapper {
       revert("DO_NOT_SEND_ETHER");
     }
   }
+  /**
+  * @notice Set whether the contract is paused
+  * @dev Only callable by owner
+  *
```
Comment (poiresel) [inline source/wrapper/contracts/Wrapper.sol]: remove blank line.

File: source/wrapper/test/Wrapper-unit.js [hunk 4]
```diff
@@ -329,4 +331,68 @@ contract('Wrapper Unit Tests', async accounts => {
...
+        wrapper.swap(order, {
+          value: 0,
+          from: mockSender,
+        })
+      )
+    })
+
+    it('Test self destruct when paused', async () => {
+      await wrapper.setPausedStatus(true)
+      await passes(wrapper.killContract(owner))
```
Comment (alicevhenshaw) [inline source/wrapper/test/Wrapper-unit.js]: You can actually check a self destruct more thoroughly as follows.
```
await wrapper.killContract(owner, { from: owner })

let contractCode = await web3.eth.getCode(wrapper.address)
equal(contractCode, '0x', 'contract did not self destruct')
```

Comment (alicevhenshaw) [reply source/wrapper/test/Wrapper-unit.js]: The way self destruct works is literally removing the contract code from existence at the address - so you can check that its gone

File: source/wrapper/test/Wrapper-unit.js (hunk 2 above)
Comment (ejwessel) [reply source/wrapper/test/Wrapper-unit.js]: Interesting. I didn't realize this was added in their contracts.

File: source/wrapper/contracts/Wrapper.sol [hunk 5]
```diff
@@ -20,18 +20,23 @@ pragma experimental ABIEncoderV2;
 import "@airswap/swap/contracts/interfaces/ISwap.sol";
 import "@airswap/tokens/contracts/interfaces/IWETH.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
```
Comment (poiresel) [inline source/wrapper/contracts/Wrapper.sol]: remove this import, it's not used

**Pull request merged**